app/src/main/java/com/example/composetemplate/presentation/screens/main_screens/MainViewModel.kt

```diff
@@ -25,6 +34,20 @@ class MainViewModel(private val repository: TestRepository) : ViewModel() {
         }
     }
 
+    fun writeDataToDS() {
+        viewModelScope.launch {
+            application.readValue(DS_TEST_KEY,DataStoreType.STRING).collect{ data ->
+                (data as? String)?.let {
```

Comment:
In this way, you write the first tome to data store then you only get the value instead of overwriting it no? we need fun to write or overwrite the value and get a different fun to get the data

Reply:
Yes, I just wanted to check the implementation. The function saves the data once and then just reads it (if the data exists). When we will implement a real API and UI, I will find a real usage for the DS logic.

Reply:
Yes I did